src/containers/Resources/index.jsx

```diff
@@ -1,18 +1,114 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import SearchBar from "../../components/SearchBar";
 import NgoList from "../../components/NgoList";
 import AdvancedFilters from "../../components/AdvancedFilters";
 import PagesList from "../../components/PagesList";
 
+import database from "../../firebaseConfig";
+
+// Fetch all ngos from the database
+const fetchNgos = (setNgos) => {
+  database
+    .collection("ngosEn")
+    .get()
+    .then((querySnapshot) => {
+      querySnapshot.forEach((doc) => {
+        setNgos((prevState) => [
+          ...prevState,
+          { id: doc.id, ngoData: doc.data() },
+        ]);
+      });
+    })
+    .catch((error) => {
+      alert(`Error getting documents ===> ${error.message}`);
+    });
+};
+
+const checkIfIsIncluded = (option, ngoValue) => {
+  if (!option.includes(ngoValue)) {
+    option.push(ngoValue);
+  }
+};
+
+// Advance Filters component's functionalities - START
+// OUTPUT ==> [{city:["ankara","Sincan"]}, {service: ["health care", "finance"]}]
+const getFilterOptions = (ngos) => {
+  const filterOptions = { cities: [], services: [], ratings: [], names: [] };
+
+  for (let i = 0; i < ngos.length; i++) {
+    checkIfIsIncluded(filterOptions.cities, ngos[i].city);
+    checkIfIsIncluded(filterOptions.services, ngos[i].service);
+    checkIfIsIncluded(filterOptions.ratings, ngos[i].rating);
+    checkIfIsIncluded(filterOptions.names, ngos[i].name);
+    return filterOptions;
+  }
+};
+
+const applyAdvancedFilters = (filterOptions, ngos, setFilteredNgos) => {
+  setFilteredNgos([]);
+  for (let i = 0; i < ngos.length; i++) {
+    const isAllOptionsIncluded =
+      filterOptions.cities.includes(ngos[i].city) &&
```

Comment:
what if there's no filter options to start with ?

Reply:
We don't show any filter option and the filteredNgos state would be [].